integration_tests/vec_01.py

```diff
@@ -0,0 +1,9 @@
+from ltypes import f64
+from numpy import empty
+
+a: f64[9216] = empty(9216)
+b: f64[9216] = empty(9216)
+b[:] = 5.0
+a[:] = b[:]
+
+print(a)
```

Comment:
Does this test make sense to you?

Reply:
Another example:
```
for i in range(9216):
    A[i] = 4*B[i]+C[i]
```
Turns into:
```
for i in range(0,n,8):
    vector_fma(8, A[i*8:(i+1)*8], 4, B[i*8:(i+1)*8], C[i*8:(i+1)*8])
```

Reply:
Well, I avoided the loop because `visit_For` hasn't been implemented yet in AST to ASR transition. I can implement `visit_For` provided if its okay with you. We will just go out of context for a little while.

Other than implementing `visit_For` we will also need to implement a pass which converts `For` into `While` so that LLVM handles only one type of loop all the time.

Reply:
Or we can just start with implementing vectorisation in LFortran and then come to LPython. What do you say?

Reply:
I am remembering now. The next step that we also need is to then unroll the loop, so from:
```
for i in range(0,n,8):
    vector_copy(8, A[i*8:(i+1)*8], B[i*8:(i+1)*8])
```
into:
```
for i in range(...):
    vector_copy(8, A[...], B[...])
    vector_copy(8, A[...], B[...])
    vector_copy(8, A[...], B[...])
    vector_copy(8, A[...], B[...])
    vector_copy(8, A[...], B[...])
    vector_copy(8, A[...], B[...])
    vector_copy(8, A[...], B[...])
    vector_copy(8, A[...], B[...])
```
That is how speed is obtained. So in assembly the loop body is "thick", full of vector instructions.

Reply:
I think we have for loops already implemented in LPython, and we also have do loops -> while loop ASR pass.

Reply:
I am getting the following error,

```bash
Internal Compiler Error: Unhandled exception
Traceback (most recent call last):
  File "/Users/czgdp1807/lpython_project/lpython/src/bin/lpython.cpp", line 842
    return emit_asr(arg_file, runtime_library_dir,
  File "/Users/czgdp1807/lpython_project/lpython/src/bin/lpython.cpp", line 168
    r = LFortran::LPython::python_ast_to_asr(al, *ast, diagnostics, true,
  File "/Users/czgdp1807/lpython_project/lpython/src/lpython/semantics/python_ast_to_asr.cpp", line 3752
    auto res = symbol_table_visitor(al, *ast_m, diagnostics, main_module,
  File "/Users/czgdp1807/lpython_project/lpython/src/lpython/semantics/python_ast_to_asr.cpp", line 2361
    SymbolTableVisitor v(al, nullptr, diagnostics, main_module, ast_overload, parent_dir);
  File "/Users/czgdp1807/lpython_project/lpython/src/lpython/semantics/python_ast_to_asr.cpp", line 2104
    visit_stmt(*x.m_body[i]);
  File "/Users/czgdp1807/lpython_project/lpython/src/lpython/python_ast.h", line 1862
    void visit_stmt(const stmt_t &b) { visit_stmt_t(b, self()); }
  File "/Users/czgdp1807/lpython_project/lpython/src/lpython/python_ast.h", line 1738
    case stmtType::AnnAssign: { v.visit_AnnAssign((const AnnAssign_t &)x); return; }
  File "/Users/czgdp1807/lpython_project/lpython/src/lpython/python_ast.h", line 1871
    void visit_For(const For_t & /* x */) { throw LFortran::LFortranException("visit_For() not implemented"); }
LFortranException: visit_For() not implemented
```

Reply:
Ah I see. Its implemented in the BodyVisitor.

Reply:
The error you are getting is due to #649, which is a separate issue.

Reply:
>  But since we don't have for loops -> while loop ASR pass.

We do: For loops are represented as ASR `DoLoop` node, which then gets transformed to `While` for the LLVM backend.

Reply:
Okay. Makes sense.

Reply:
Made some changes. Feel free to take a look and let me know if any changes are needed. I will implement a vectorisation pass in the next step.

Reply:
Looks good!